miden-lib/src/accounts/faucets/mod.rs

```diff
@@ -33,8 +33,12 @@ pub struct BasicFungibleFaucet {
 impl BasicFungibleFaucet {
     // CONSTANTS
     // --------------------------------------------------------------------------------------------
-    const MAX_MAX_SUPPLY: u64 = (1 << 63) - 1;
-    const MAX_DECIMALS: u8 = 12;
+
+    /// The maximum value for the `max_supply` field.
+    pub const MAX_MAX_SUPPLY: u64 = (1 << 63) - 1;
```

Comment:
We already have a similar constant in `miden-base`: `FungibleAsset::MAX_AMOUNT` - so, not sure we need to have it repeated here (and we can actually remove it from here entirely).

`MAX_DECIMALS` is not something we have in `miden-base` though and, unlike `MAX_AMOUNT`, it is not enforced at the protocol level. But I do wonder if we should move `MAX_DECIMALS` there as well.

Alternatively, we could keep both here, but then I'd change `MAX_MAX_SUPPLY` to something like:
```rust
pub const MAX_MAX_SUPPLY: u64 = FungibleAsset::MAX_AMOUNT;
```

Reply:
Since `MAX_DECIMALS` is not protocol enforced, I don't think that belongs in `FungibleAsset`. Maybe the best is to keep it in `BasicFungibleFaucet`.

On the other hand, I'm removing the `MAX_MAX_SUPPLY` and replacing it with `FungibleAsset::MAX_AMOUNT` directly.